src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -27,12 +28,21 @@ internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactor
             IOptions<DurableTaskOptions> options,
             IConnectionStringResolver connectionStringResolver,
             INameResolver nameResolver,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            IPlatformInformationService platformInfo = null)
         {
             this.options = options.Value;
             this.nameResolver = nameResolver;
             this.loggerFactory = loggerFactory;
             this.azureStorageOptions = new AzureStorageOptions();
+            this.inConsumption = platformInfo != null ? platformInfo.InConsumption() : false;
+
+            if (this.inConsumption)
+            {
+                this.azureStorageOptions.ControlQueueBufferThreshold = 32;
+                this.options.MaxConcurrentOrchestratorFunctions = 5;
```

Comment:
Other settings we may want to edit include `this.options.MaxConcurrentActivityFunctions`.

It's also worth adjusting `this.azureStorageOptions.ControlQueueBatchSize`. Currently that defaults to 32, which means we are going to get a very start/stop pre-fetch behavior given the current control queue buffer threshold. Maybe the batch size should be some fraction of the controlqueuebuffertreshold by default?

Reply:
I've included performance data based on setting `MaxConcurrentActivityFunctions` to10 and `ControlQueueBatchSize` to 0.25*`ControlQueueBufferThreshold`. These results are named PRv3 and PRv2 respectively. From what I gather so far (at the time of writing the results for benchmark D are still to come) these didn't make a big difference in performance.

Since I didn't see a noticeable effect by changing those settings, I'm thinking it might be a good idea to revert back to the settings of PRv1 and to reconsider further tweaking of `MaxConcurrentActivityFunctions` and `ControlQueueBatchSize` for a future PR. What do you think, @ConnorMcMahon?

We can re-consider our decision once the results for benchmark D arrive in a few hours, but I have no reason to think they'll be surprising.